### PR TITLE
Added BitTorrent Sync

### PR DESCRIPTION
**Pull request by Rohan62442**
See #94

### CODE REVIEW

File: plugins/bittorrentsync.plugin/metadata.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	"scripts": {
+		"exec": {
+			"label": "Install",
+			"command": "sh install.sh"
+		},
+		"undo": {
+			"label": "Remove",
+			"command": "sh uninstall.sh"
+		},
+		"status": { "command": "bash -c 'test -e ~/.local/share/bittorrent-sync/btsync'" }
```
Comment (satya164) [inline plugins/bittorrentsync.plugin/metadata.json]: You can just use `test -e ~/.local/share/bittorrent-sync/btsync`

**Comment by satya164** [comment]
This installs the app for the current user only, while other plugins install apps for all users. May be we could do that here too?

Fedy provides a helper command `run-as-root`, which is just an alias for `pkexec` to show a GUI password prompt.

Note that `pkexec` doesn't keep the working directory, so when trying to run a script situated in the plugin directory as root, you need to do it like `run-as-root -s install.sh`

Thanks :)

**Comment by Rohan62442** [comment]
If you check the issue I had in the previous versions of Fedy, it was recommended that BitTorrent Sync not be installed as root. Now I no longer remember why... :(

I will check the root installation (I knew about `run-as-root`)

**Comment by satya164** [comment]
I did some search and they say not to run btsync as root. Didn't find anything about installing as root though.

I also came across this wiki - https://wiki.archlinux.org/index.php/BitTorrent_Sync

The wiki also installed it as root.

**Comment by Rohan62442** [comment]
OK, then. I will modify this.

**Comment by Rohan62442** [comment]
I have not tested this though..

File: plugins/bittorrentsync.plugin/install.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+if [[ ! -f "$FILE" ]]; then
+	exit 1
+fi
+
+mkdir -p "/opt/bittorrent-sync"
+tar -xf "$FILE" -C "/opt/bittorrent-sync"
+ln -sf "/opt/bittorrent-sync/btsync" "/usr/bin/btsync"
+
+btsync --dump-sample-config > /etc/btsync.conf
+systemctl enable btsync
```
Comment (satya164) [inline plugins/bittorrentsync.plugin/install.sh]: I don't think you can enable or start services without installing them

**Comment by Rohan62442** [comment]
Well, for me, the root install does not work.

**Comment by Jobava** [comment]
Another way to install it may be to create a separate user and group for that particular app with its own permissions and selinux rules, as it's sometimes done for apache.

**Pull request closed without merging**